nixos/modules/services/x11/desktop-managers/elementary.nix

```diff
@@ -0,0 +1,201 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.xserver.desktopManager.elementary;
+
+  nixos-gsettings-desktop-schemas = pkgs.elementary.elementary-gsettings-schemas.override {
+    extraGSettingsOverridePackages = cfg.extraGSettingsOverridePackages;
+    extraGSettingsOverrides = cfg.extraGSettingsOverrides;
+  };
+
+in
+
+{
+  options = {
+
+    services.xserver.desktopManager.elementary = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable the elementary desktop manager";
+      };
+
+      sessionPath = mkOption {
+        default = [];
+        example = literalExample "[ pkgs.gnome3.gpaste ]";
+        description = ''
+          Additional list of packages to be added to the session search path.
+          Useful for GSettings-conditional autostart.
+
+          Note that this should be a last resort; patching the package is preferred (see GPaste).
+        '';
+        apply = list: list ++
+        [
+          pkgs.elementary.elementary-dpms-helper
+          pkgs.elementary.elementary-settings-daemon # TODO: I shouldn't need to do this
+          pkgs.elementary.pantheon-agent-geoclue2
+          pkgs.gnome3.evolution-data-server
```

Comment:
I had to add this because `switchboard-plug-datetime` would crash with
```
Settings schema 'org.gnome.Evolution.DefaultSources' is not installed
```
when selected.

Didn't find a direct reference to that key in the source though.

Reply:
I think that comes form `e-d-s` itself, we will need to patch it.

Reply:
Here https://github.com/GNOME/evolution-data-server/blob/4758aed465e74a993703bf8248da72a3e7107838/src/libedataserver/e-source-registry.c#L1752?

Reply:
Looks like it.

Reply:
I've kept `pantheon-agent-geoclue2` in session path because it truly was a last resort as no amount of patching would produce a functional agent. Same for dpms-helper I think.